src/shared/sam/activation.ts

```diff
@@ -138,32 +142,75 @@ async function activateCodeLensProviders(
-            csLensProvider.CSHARP_ALLFILES,
-            await codelensUtils.makeCSharpCodeLensProvider()
-        )
+        vscode.commands.registerCommand('aws.toggleSamCodeLenses', () => {
+            configuration.readSetting<boolean>(codelensUtils.STATE_NAME_SUPPRESS_CODELENSES)
+                ? configuration.writeSetting(
+                      codelensUtils.STATE_NAME_SUPPRESS_CODELENSES,
+                      false,
+                      vscode.ConfigurationTarget.Global
+                  )
+                : configuration.writeSetting(
+                      codelensUtils.STATE_NAME_SUPPRESS_CODELENSES,
+                      true,
+                      vscode.ConfigurationTarget.Global
+                  )
+        })
     )
 
+    /**
+     * This should only be callable when the current editor is a JS, Python, C#, or Java editor.
+     * Error blocks should never be triggered, but messages will be provided regardless (I guess a user could this to a keyboard shortcut...)
+     */
     disposables.push(
-        vscode.languages.registerCodeLensProvider(
-            javaLensProvider.JAVA_ALLFILES,
-            await codelensUtils.makeJavaCodeLensProvider()
-        )
+        vscode.commands.registerCommand('aws.addSamDebugConfig', async () => {
+            const activeEditor = vscode.window.activeTextEditor
+            if (!activeEditor) {
+                getLogger().error(`aws.addSamDebugConfig was called without an active text editor`)
+                vscode.window.showErrorMessage(
+                    'AWS.pickDebugHandler.noEditor',
+                    'Toolkit could not find an active editor'
+                )
+
+                return
+            }
+            const document = activeEditor.document
+            let provider: codelensUtils.OverridableCodeLensProvider | undefined
+            if (document.languageId === 'javascript') {
+                provider = tsCodeLensProvider
+            } else if (document.languageId === 'python') {
+                provider = pyCodeLensProvider
+            } else if (document.languageId === 'csharp') {
+                provider = csCodeLensProvider
+            } else if (document.languageId === 'java') {
+                provider = javaCodeLensProvider
+            } else {
+                getLogger().error(
+                    `aws.addSamDebugConfig called on a document with an invalid language: ${document.languageId}`
+                )
+                vscode.window.showErrorMessage(
+                    localize(
+                        'AWS.pickDebugHandler.invalidLanguage',
+                        'Toolkit cannot detect handlers in language: {0}',
+                        document.languageId
+                    )
+                )
+
+                return
+            }
+
+            const lenses =
+                (await provider.provideCodeLenses(document, new vscode.CancellationTokenSource().token, true)) ?? []
```

Comment:
A new CancellationToken is created here but we don't really do anything with it. Are there plans to add a timeout perhaps?

Reply:
No. We just need one to fill out the arg. We don't even do anything with it on the other end (we'd have to have a promise race). Unknown if there's a way to cancel a CodeLensProvider in progress anyway.